tests/late_joiners.py

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+import os
+import getpass
+import time
+import logging
+import multiprocessing
+import shutil
+from random import seed
+import infra.ccf
+import infra.proc
+import infra.notification
+import infra.net
+import e2e_args
+from threading import Timer
+import random
+import contextlib
+
+from loguru import logger as LOG
+
+# 256 is the number of most recent messages that PBFT keeps in memory before needing to replay the ledger
+TOTAL_REQUESTS = 56
+
+
+def timeout(node, suspend, election_timeout):
```

Comment:
Is `suspend` always `True`?

Reply:
No (see line 31 below) it is then set to `False` so it will timeout once and suspend the node, then the new timer will time out and resume the node